.github/workflows/build-pr-image.yml

```diff
@@ -0,0 +1,97 @@
+
+name: Build PR Docker Image
+
+on:
+  push:
+    branches:
+      - "main"
+    paths-ignore:
+      - "**.md"
+  pull_request:
+    branches:
+      - "*"
+    paths-ignore:
+      - "**.md"
+env:
+    AWS_REGION: us-east-1
+
+jobs:
+    build:
+      name: build
+      permissions:
+        id-token: write
+        contents: read
+      runs-on: ubicloud-standard-16
+      # runs-on: ubuntu-latest
+      steps:
+        - name: Remove unused tools
+          run: |
+            sudo rm -rf /usr/share/dotnet
+            sudo rm -rf /opt/ghc
+            sudo rm -rf "/usr/local/share/boost"
+            sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+  
+        - name: Clone the current repo
+          uses: actions/checkout@v4
+          with:
+            fetch-depth: 0
+  
+        - name: Setup Rust Toolchain
+          uses: dtolnay/rust-toolchain@master
+          with:
+            toolchain: nightly-2023-12-24
+            
+        - uses: Swatinem/rust-cache@v2
+          with:
+            cache-on-failure: true
+        - name: Install Protoc
+          uses: arduino/setup-protoc@v2
+          with:
+            version: "21.12"
+        - uses: actions/setup-node@v4
+          with:
+            node-version: 18
+        
+        - name: Build frontend code
+          run: cd web && npm install && npm run build
+        
+        - name: Build and run binary
+          run: |
+            mkdir -p bin && \
+            cargo build --release --features mimalloc --target x86_64-unknown-linux-gnu && \
```

Comment:
If it's an image a PR why do a release build instead of debug build?

Reply:
We do deploy it as a release build several times actually in the internal infra, as compared to what we did with API- testing one.